examples/oapp-solana/programs/my_oapp/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+mod errors;
+mod instructions;
+mod msg_codec;
+mod state;
+
+use anchor_lang::prelude::*;
+use instructions::*;
+use oapp::{endpoint::MessagingFee, endpoint_cpi::LzAccount, LzReceiveParams};
+use solana_helper::program_id_from_env;
+use state::*;
+
+declare_id!(anchor_lang::solana_program::pubkey::Pubkey::new_from_array(program_id_from_env!(
+    "MYOAPP_ID",
+    "HFyiETGKEUS9tr87K1HXmVJHkqQRtw8wShRNTMkKKxay"
+)));
```

Comment:
The current program ID `HFyiETGKEUS9tr87K1HXmVJHkqQRtw8wShRNTMkKKxay` is already in use by the OmniCounter program. Consider generating a new program ID for this OApp